### PR TITLE
feat(Search): Improved search bar

### PR DESCRIPTION
**Pull request by KegenGuyll**

### CODE REVIEW

File: components/navigation/navSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+
+interface ChipProps extends SearchTopic {
+  onClick: () => void
+  compact?: boolean
+}
+
+const localStorageSearch = 'searches'
+
+const NavSearchChip: FunctionComponent<ChipProps> = ({label, icon, onClick, compact}: ChipProps) => (
```
Comment (KegenGuyll) [inline components/navigation/navSearch.tsx:39]: Break out into its own file

File: components/navigation/navSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+  const handleClearResults = () => {
+    localStorage.removeItem(localStorageSearch)
+
+    setPrevSearches([])
+  }
+
+
+  return (
+    <div
```
Comment (KegenGuyll) [inline components/navigation/navSearch.tsx:138]: come up with a more mobile friendly design

File: pages/search.tsx [hunk 3]
```diff
@@ -161,22 +191,46 @@ const Search: NextPage<Props> = ({ q }: Props) => {
...
+      store.dispatch(findMovieYear.initiate({
+        page,
+        primary_release_year: Number(q),
+        sort_by: SortBy.VoteCountDesc
+      }))
     }
 
+    // if (typeof q === 'string' && topic) {
+    //   store.dispatch(search.initiate({q, topic}));
+    // } else if (typeof q === 'string') {
```
Comment (KegenGuyll) [inline pages/search.tsx:223]: remove comments

File: redux/api/index.ts [hunk 4]
```diff
@@ -214,10 +214,18 @@ export const tmrevApi = createApi({
         url: `/movie/watched/${userId}`,
       }),
     }),
-    search: builder.query<SearchResponse, string>({
-      query: (data) => ({
-        url: `/search?q=${data}`,
-      }),
+    search: builder.query<SearchResponse, { q?: string, topic?: string }>({
+      query: ({ q, topic }) => {
```
Comment (KegenGuyll) [inline redux/api/index.ts:218]: revert

**Review by ayronman99 — APPROVED** [review]
Merging to dev

**Pull request merged**